share-on-mastodon.js

```diff
@@ -197,120 +208,143 @@ export class shareOnMastodon extends HTMLElement {
+      this.shadowRoot.querySelector("#instance-input").value = "https://" + val;
     }
+    this.shadowRoot.querySelector("#instance-input").checkValidity();
   }
 
-  share_on_mastodon() {
+  shareOnMastodon() {
     // Set instance from the modal text input
-    let instance = this.shadowRoot.querySelector('#instanceInput').value;
+    let instance = this.shadowRoot.querySelector("#instance-input").value;
 
     // check, format, and cache instance
     if (instance) {
       // Handle URL formats
-      if ( !instance.startsWith("https://") && !instance.startsWith("http://") )
+      if (!instance.startsWith("https://") && !instance.startsWith("http://"))
         instance = "https://" + instance;
 
       // Handle ending slash
-      if ( !instance.endsWith("/") )
-        instance = instance + "/";
+      if (!instance.endsWith("/")) instance = instance + "/";
 
       // Cache the instance/domain for future requests
-      localStorage['mastodon_instance'] = instance;
-    };
+      localStorage.setItem("mastodonInstance", instance);
+    }
 
-    // If the share_title attribute is set use it. Otherwise, use the page title
+    // If the share-title attribute is set use it. Otherwise, use the page title
     let shareTitle = this.pageTitle;
-    if (this.getAttribute('share_title')) {
-      shareTitle = this.getAttribute('share_title')  + "\n";
+    if (this.getAttribute("data-share-title")) {
+      shareTitle = this.getAttribute("data-share-title") + "\n";
     } else {
-      shareTitle = docTitle  + "\n";
-    };
+      shareTitle = docTitle + "\n";
+    }
 
-    // If the share_description attribute is set use it. Otherwise, only use this for formating
-    let desc = this.getAttribute('share_description');
+    // If the share-description attribute is set use it. Otherwise, only use this for formating
+    let desc = this.getAttribute("data-share-description");
     if (desc) {
       desc = desc + "\n\n";
     } else {
       desc = "\n";
-    };
+    }
 
     // If the hashtags attribute is set use it. Otherwise, set it to an empty string
-    let hashtags = this.getAttribute('hashtags');
+    let hashtags = this.getAttribute("data-hashtags");
     if (hashtags) {
       hashtags = "\n\n" + hashtags;
     } else {
       hashtags = "";
-    };
+    }
 
     // If the author attribute is set use it. Otherwise, set it to an empty string
-    let author = this.getAttribute('author');
+    let author = this.getAttribute("data-author");
     if (author) {
       author = "\n\n" + author;
     } else {
       author = "";
-    };
+    }
 
     // Get the current page's URL
     const url = window.location.href;
 
     // Create the Share URL
-    const mastodon_url = instance + "share?text=" + encodeURIComponent(shareTitle + desc + url + hashtags + author);
+    const mastodonUrl =
+      instance +
+      "share?text=" +
+      encodeURIComponent(shareTitle + desc + url + hashtags + author);
```

Comment:
You can also use the SearchParamAPI to build out urls with encoded query params. It can by handy: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams

Reply:
Cool, I'll read through the documentation and see if I can't use it to clean this up a bit.

Reply:
Ok, I'm not entirely sure I'm using this to it's potential but I am using it now 😅